dask/array/creation.py

```diff
@@ -73,7 +73,18 @@ def empty_like(a, dtype=None, order="C", chunks=None, name=None, shape=None):
 
     a = asarray(a, name=False)
     shape, chunks = _get_like_function_shapes_chunks(a, chunks, shape)
-    return empty(shape, dtype=(dtype or a.dtype), order=order, chunks=chunks, name=name)
+    result = empty(
+        shape,
+        dtype=(dtype or a.dtype),
+        order=order,
+        chunks=chunks,
+        name=name,
+        meta=a._meta,
+    )
+    shape_meta = result._meta.shape
+    result._meta = a._meta  # preserves arraytype in repr_html
+    result._meta.shape = shape_meta  # preserves ndim if array reshaped
```

Comment:
I'm curious, is there a nice way to have the underlying functions handle this?  It seems like they should probably respect the `meta=` keyword.

Reply:
The underlying functions are `wrap` and `broadcast_trick` in `dask/array/wrap.py`. Maybe there's a way to get the wrapping functions to handle this? 

I haven't totally untangled how the wrapping works in my own mind just yet.

Reply:
> I haven't totally untangled how the wrapping works in my own mind just yet.

Yeah... it's also somewhat tangled in my mind too these days :)